File: invoicer-backend/src/main/java/com/chronica/invoicer/data/dto/InvoiceDTO.java
package com.chronica.invoicer.data.dto;

import com.chronica.invoicer.data.EntityDTO;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class InvoiceDTO extends EntityDTO {
    private String symbol;
    private Date creationDate;
    private Date saleDate;
    private Date paymentDate;
    private CompanyDTO seller;
    private CompanyDTO customer;
    private List<InvoiceItemDTO> invoiceItems;
    private InvoicePriceDTO invoicePrice;
}

File: invoicer-ui/src/invoicer/model/InvoiceDTO.ts
import {CompanyDTO} from "../../company/model/CompanyDTO";
import {ProductDTO} from "../../product/model/ProductDTO";
import {EntityDTO} from "../../shared/model/EntityDTO";

export interface InvoiceDTO extends EntityDTO {
    symbol?: string,
    creationDate?: string,
    saleDate?: string,
    paymentDate?: string,
    seller?: CompanyDTO,
    customer?: CompanyDTO,
    invoicePrice?: InvoicePriceDTO,
    invoiceItems?: InvoiceItemDTO[],
}

export interface InvoiceItemDTO extends EntityDTO {
    invoicePosition?: number;
    quantity?: number;
    discount?: number;
    archivalProduct?: ProductDTO;
    partialPrice?: number;
}

export interface InvoicePriceDTO extends EntityDTO {
    netAmount?: number;
    vatAmount?: number;
    brutAmount?: number;
    invoiceItems?: InvoiceItemDTO[];
}